src/tryptag/tryptools/tryptools.py

```diff
@@ -246,9 +255,17 @@ def cell_morphology_analysis(cell_image) -> dict:
           distance.append(distance[-1] + root2)
         else:
           distance.append(distance[-1] + 1)
+
+      distances = midline_analysis["mat_distances"]
+      start = midline_analysis["anterior"]
+      end = midline_analysis["posterior"]
+      distance_ant = distances[start[0], start[1]]
+      distance_post = distances[end[0], end[1]]
+
       midline_analysis.update({
         "distance": distance,
-        "length": distance[-1]
+        "length": distance[-1],
+        "corrected_length": distance[-1] + distance_ant + distance_post
```

Comment:
I think this measure is clearly better than the naive skeleton length, so just replace length with the new measure.